lib/src/main/java/com/dynatrace/file/util/WatchServiceBasedFilePoller.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2022 Dynatrace LLC
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dynatrace.file.util;
+
+import static java.nio.file.StandardWatchEventKinds.ENTRY_CREATE;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_MODIFY;
+
+import com.sun.nio.file.SensitivityWatchEventModifier;
+import java.io.IOException;
+import java.nio.file.*;
+import java.util.Collections;
+import java.util.List;
+
+class WatchServiceBasedFilePoller extends AbstractFilePoller {
```

Comment:
This one didn't change from what we had before, right? Just a new file.

Reply:
It is very similar, yes.